File: app/src/main/java/com/movieapplication/model/tvseries/Result.java
package com.movieapplication.model.tvseries;


import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import java.util.List;

import javax.annotation.Generated;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Generated("jsonschema2pojo")
public class Result implements Parcelable {

    @SerializedName("page")
    @Expose
    private Integer page;
    @SerializedName("results")
    @Expose
    private List<Tvserie> results;
    @SerializedName("total_pages")
    @Expose
    private Integer totalPages;
    @SerializedName("total_results")
    @Expose
    private Integer totalResults;

    // Parcelable Creator
    public final static Creator<Result> CREATOR = new Creator<Result>() {
        @Override
        public Result createFromParcel(Parcel parcel) {
            return new Result(parcel);
        }

        @Override
        public Result[] newArray(int i) {
            return new Result[i];
        }
    };


    public Result() {
    }

    public Result(Parcel in) {
        this.page = (Integer) in.readValue(Integer.class.getClassLoader());
        in.readList(this.results, (Tvserie.class.getClassLoader()));
        this.totalPages = (Integer) in.readValue(Integer.class.getClassLoader());
        this.totalResults = (Integer) in.readValue(Integer.class.getClassLoader());
    }


    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public List<Tvserie> getResults() {
        return results;
    }

    public void setResults(List<Tvserie> results) {
        this.results = results;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(@NonNull Parcel parcel, int i) {
        parcel.writeValue(page);
        parcel.writeValue(results);
        parcel.writeValue(totalPages);
        parcel.writeValue(totalResults);
    }
}

File: app/src/main/java/com/movieapplication/model/tvseries/Result.kt
package com.movieapplication.model.tvseries

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import javax.annotation.Generated

@Generated("jsonschema2pojo")
class Result : Parcelable {
    @SerializedName("page")
    @Expose
    var page: Int? = null

    @SerializedName("results")
    @Expose
    var results: List<Tvserie?>? = null

    @SerializedName("total_pages")
    @Expose
    var totalPages: Int? = null

    @SerializedName("total_results")
    @Expose
    var totalResults: Int? = null

    constructor() {}
    constructor(`in`: Parcel) {
        page = `in`.readValue(Int::class.java.classLoader) as Int?
        `in`.readList(results!!, Tvserie::class.java.classLoader)
        totalPages = `in`.readValue(Int::class.java.classLoader) as Int?
        totalResults = `in`.readValue(Int::class.java.classLoader) as Int?
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeValue(page)
        parcel.writeValue(results)
        parcel.writeValue(totalPages)
        parcel.writeValue(totalResults)
    }

    companion object {
        // Parcelable Creator
        val CREATOR: Creator<Result?> = object : Creator<Result?> {
            override fun createFromParcel(parcel: Parcel): Result? {
                return Result(parcel)
            }

            override fun newArray(i: Int): Array<Result?> {
                return arrayOfNulls(i)
            }
        }
    }

 object CREATOR : Creator<Result> {
        override fun createFromParcel(parcel: Parcel): Result {
            return Result(parcel)
        }

        override fun newArray(size: Int): Array<Result?> {
            return arrayOfNulls(size)
        }
    }
}